tests/sentry/relay/config/test_metric_extraction.py

```diff
@@ -1261,3 +1585,46 @@ def test_get_metric_extraction_config_with_no_spec(default_project: Project) ->
         assert config
         assert len(config["metrics"]) == 1
         assert config["metrics"][0].get("condition") is None
+
+
+def _metric_spec(tags: list[dict[str, str]]) -> dict[str, Any]:
+    return {
+        "category": "transaction",
+        "condition": {"name": "event.duration", "op": "gte", "value": 10.0},
+        "field": None,
+        "mri": "c:transactions/on_demand@none",
+        "tags": tags,
+    }
+
+
+@django_db_all
+def test_include_environment_for_widgets(default_project: Project) -> None:
+    aggr = "count()"
+    env_tag = {"field": "event.environment", "key": "environment"}
+    query = "transaction.duration:>=10"
+
+    with Feature([ON_DEMAND_METRICS, ON_DEMAND_METRICS_WIDGETS]):
+        create_widget([aggr], query, default_project)
+        config = get_metric_extraction_config(default_project)
+        # Because we have two specs we will have two metrics.
+        # The second spec includes the environment tag as part of the query hash.
+        assert config and config["metrics"] == [
+            _metric_spec([{"key": "query_hash", "value": "f1353b0f"}, env_tag]),
+            _metric_spec([{"key": "query_hash", "value": "4fb5a472"}, env_tag]),
+        ]
+
+        # Using environment in the columns will turn into a groupbys
+        create_widget([aggr], query, default_project, columns=["environment"], title="foo")
+        config = get_metric_extraction_config(default_project)
+        assert config and config["metrics"] == [
+            _metric_spec([{"key": "query_hash", "value": "f1353b0f"}, env_tag]),
+            _metric_spec([{"key": "query_hash", "value": "4fb5a472"}, env_tag]),
+        ]
+
+        create_alert(aggr, query, default_project)
+        config = get_metric_extraction_config(default_project)
+        # We loose the env in the tags because the alert spec is preferred over the widget spec
+        assert config and config["metrics"] == [
+            _metric_spec([{"key": "query_hash", "value": "f1353b0f"}]),
```

Comment:
@k-fish do you want me to fix that we pick up the spec from the alert instead of the widget one which includes the environment in the tags?

Should the alert spec include the env in the tags?

Reply:
Let's have both tests:
1. A test without environment in columns, they should be 2 different specs with 2 different hashes, no collision.
2. A test with environment in alert, but not in widget. This should make 2 identical specs with the same hash. Which one we pick here is irrelevant.